lib/patchestry/intrinsics/patch_utils.c

```diff
@@ -0,0 +1,320 @@
+    
+    return state;
+}
+
+bool __patchestry_patch_set_state(const char* patch_name, patchestry_patch_state_t state) {
+    if (patch_name == NULL) {
+        return false;
+    }
+    
+    pthread_mutex_lock(&patch_mutex);
+    patch_entry_t* entry = find_patch_entry(patch_name);
+    if (entry) {
+        entry->state = state;
+    }
+    pthread_mutex_unlock(&patch_mutex);
+    
+    return entry != NULL;
+}
+
+bool __patchestry_patch_store_data(const char* patch_name, const void* data, size_t size) {
+    if (patch_name == NULL || data == NULL || size == 0) {
+        return false;
+    }
+    
+    pthread_mutex_lock(&patch_mutex);
+    patch_entry_t* entry = find_patch_entry(patch_name);
+    if (entry) {
+        if (entry->patch_data) {
+            free(entry->patch_data);
+        }
+        
+        entry->patch_data = malloc(size);
+        if (entry->patch_data) {
+            memcpy(entry->patch_data, data, size);
+            entry->data_size = size;
+        }
+    }
+    pthread_mutex_unlock(&patch_mutex);
+    
+    return entry != NULL && entry->patch_data != NULL;
+}
+
+void* __patchestry_patch_get_data(const char* patch_name) {
+    if (patch_name == NULL) {
+        return NULL;
+    }
+    
+    pthread_mutex_lock(&patch_mutex);
+    patch_entry_t* entry = find_patch_entry(patch_name);
+    void* data = entry ? entry->patch_data : NULL;
+    pthread_mutex_unlock(&patch_mutex);
+    
+    return data;
+}
+
+size_t __patchestry_patch_get_data_size(const char* patch_name) {
+    if (patch_name == NULL) {
+        return 0;
+    }
+    
+    pthread_mutex_lock(&patch_mutex);
+    patch_entry_t* entry = find_patch_entry(patch_name);
+    size_t size = entry ? entry->data_size : 0;
+    pthread_mutex_unlock(&patch_mutex);
+    
+    return size;
+}
+
+// =============================================================================
+// Conditional Patching Implementation
+// =============================================================================
+
+bool __patchestry_check_env_variable(const char* var_name, const char* expected_value) {
+    if (var_name == NULL) {
+        return false;
+    }
+    
+    const char* value = getenv(var_name);
+    if (value == NULL) {
+        return expected_value == NULL;
+    }
+    
+    if (expected_value == NULL) {
+        return false;
+    }
+    
+    return strcmp(value, expected_value) == 0;
+}
+
+bool __patchestry_check_build_config(const char* config_name) {
+    if (config_name == NULL) {
+        return false;
+    }
+    
+    // Check common build configurations
+    if (strcmp(config_name, "DEBUG") == 0) {
+        #ifdef DEBUG
+        return true;
+        #else
+        return false;
+        #endif
+    }
+    
+    if (strcmp(config_name, "RELEASE") == 0) {
+        #ifdef NDEBUG
+        return true;
+        #else
+        return false;
+        #endif
+    }
+    
+    return false;
+}
+
+bool __patchestry_check_target_arch(const char* arch_name) {
```

Comment:
I don't have as much of a bad feeling about strcmp here, but perhaps it could be changed for consistency too? (ffti)